debezium-server-eventhubs/src/main/java/io/debezium/server/eventhubs/BatchManager.java

```diff
@@ -34,11 +38,13 @@ public class BatchManager {
     private final HashMap<String, EventDataBatchProxy> dynamicPartitionKeyBatches = new HashMap<>();
 
     public BatchManager(EventHubProducerClient producer, String configurePartitionId,
-                        String configuredPartitionKey, Integer maxBatchSize) {
+                        String configuredPartitionKey, Integer maxBatchSize, Boolean hashMessageKey, String hashMessageFunction) {
```

Comment:
```suggestion
                        String configuredPartitionKey, Integer maxBatchSize, Optional<String> hashMessageFunction) {
```

I think that these two parameters can be merged into a single Optional one. If present hashing is used, if not hasing is not used.